From 142bfed222c11de509c8a1b808d48e8d62f02921 Mon Sep 17 00:00:00 2001
From: Zhou Guanghui <zhouguanghui1@huawei.com>
Date: Thu, 3 Nov 2022 06:41:45 +0000
Subject: [PATCH] mm/sharepool: delete unnecessary judgment
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 142bfed222c11de509c8a1b808d48e8d62f02921
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5PIA2
CVE: NA

--------------------------------

When a process is added to a group, mm->mm_users increases by one.
When a process is deleted from a group, mm->mm_users decreases by
one. It is not possible to reduce to 0 because this function is
preceded by get_task_mm.

Signed-off-by: Zhou Guanghui <zhouguanghui1@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 mm/share_pool.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index 6942084a56d3..5b55168fee08 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -1621,11 +1621,7 @@ int mg_sp_group_del_task(int pid, int spg_id)
 	list_del(&spg_node->group_node);
 	mm->sp_group_master->count--;
 	kfree(spg_node);
-	if (atomic_sub_and_test(1, &mm->mm_users)) {
-		up_write(&sp_group_sem);
-		WARN(1, "Invalid user counting\n");
-		return -EINVAL;
-	}
+	atomic_dec(&mm->mm_users);
 
 	up_write(&sp_group_sem);
 
-- 
2.33.0

